packages/dialtone-vue2/components/rich_text_editor/rich_text_editor.vue

```diff
@@ -702,48 +711,48 @@ export default {
             ...this.inputAttrs,
             class: this.inputClass,
           },
 
-          handlePaste: (_, event) => {
-            // When having link and customLink props we should maintain default paste behavior
-            if (!this.link && !this.customLink) {
-              const regex = /^https?:\/\//;
-
-              if (!event?.clipboardData) {
-                return false;
-              }
-              const pastedContent = event.clipboardData.getData('text');
-
-              // Check if the pasted content is a valid URL (starting with http:// or https://)
-              // If it's not a URL, allow the default paste behavior
-              if (!regex.test(pastedContent)) {
-                return false;
+          handlePaste: (view, event, slice) => {
+            if (!this.pasteRichText) {
+              const clipboardData = event.clipboardData || window.clipboardData;
+              const textData = clipboardData.getData('text/plain');
+
+              if (textData) {
+                // Insert as plain text only (preserving any HTML tags as text)
+                const { tr } = view.state;
+                const { from, to } = view.state.selection;
+                tr.insertText(textData, from, to);
+                view.dispatch(tr);
+                return true; // Prevent default paste behavior
               }
-
-              // If `text/html` is missing from clipboard data, it's a plain link
-              // In this case, allow the default paste behavior
-              if (!event.clipboardData.getData('text/html')) {
-                return false;
-              }
-              const htmlContent = pastedContent
-                .replace(/\n/g, '<br>') // Convert newlines to <br>
-                .replace(/ {2}/g, '&nbsp;&nbsp;'); // Convert multiple spaces
-
-              this.editor.chain().focus().insertContent(htmlContent).run();
-              return true; // Prevent the default paste behavior
             }
 
-            return false; // Allow the default paste behavior
+            return false; // default paste behavior
           },
 
           // Moves the <br /> tags inside the previous closing tag to avoid
           // Prosemirror wrapping them within another </p> tag.
           transformPastedHTML (html) {
-            return html.replace(/(<\/\w+>)((<br \/>)+)/g, '$2$3$1');
+            // Preserve line breaks by converting them to hard breaks before other transformations
+            const transformedHtml = html
+              // Convert standalone br tags to hard breaks
+              .replace(/<br\s*\/?>/gi, '<br />')
+              // Preserve line breaks at end of paragraphs and divs
+              .replace(/(<\/(?:p|div)>)\s*\n/gi, '$1<br />')
+              // Convert newlines followed by text to br tags
+              .replace(/\n(?=\S)/g, '<br />');
+
+            // Then apply the original transformation
+            return transformedHtml.replace(/(<\/\w+>)((<br \/>)+)/g, '$2$3$1');
```

Comment:
This custom HTML transformations might cause issues in the future, but for now, they're apparently working correctly, let's just keep an eye on this.